qt/python/mantidqt/mantidqt/widgets/codeeditor/interpreter.py

```diff
@@ -379,19 +380,28 @@ def _on_exec_success(self, task_result):
     def _on_exec_error(self, task_error):
         exc_type, exc_value, exc_stack = task_error.exc_type, task_error.exc_value, task_error.stack
         exc_stack = traceback.extract_tb(exc_stack)[self.MAX_STACKTRACE_LENGTH :]
+
+        lineno = -1  # Default lineno in case of failure to retrieve it
+
         if hasattr(exc_value, "lineno"):
-            lineno = exc_value.lineno + self._code_start_offset
-        elif exc_stack is not None:
+            if exc_value.lineno is not None:
+                lineno = exc_value.lineno + self._code_start_offset
+        elif exc_stack:
             try:
-                lineno = exc_stack[0].lineno + self._code_start_offset
+                if exc_stack[0].lineno is not None:
+                    lineno = exc_stack[0].lineno + self._code_start_offset
             except (AttributeError, IndexError):
                 # Python 2 fallback
                 try:
-                    lineno = exc_stack[-1][1] + self._code_start_offset
+                    if exc_stack[-1][1] is not None:  # Ensure the last stack trace entry has a valid lineno
+                        lineno = exc_stack[-1][1] + self._code_start_offset
                 except IndexError:
-                    lineno = -1
-        else:
-            lineno = -1
+                    pass
+
+        # Log an error message if the XML is malformed or there's an issue with parsing
+        if exc_type is SyntaxError or "XML" in str(exc_value):
+            mantid.kernel.logger.error(f"Malformed XML detected. Error at line {lineno if lineno != -1 else 'unknown'}: {exc_value}")
```

Comment:
I think this error log is not necessary, the line below which writes using the error formatter creates 
`xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 1, column 0` which I think is descriptive enough on its own.
With this log you get the same info twice
```
Malformed XML detected. Error at line unknown: not well-formed (invalid token): line 1, column 0
xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 1, column 0
```

Reply:
Would it be better if it did this?

`mantid.kernel.logger.error(f"Malformed XML detected. Make sure XML input is correct ")`

Reply:
I think you don't need it at all since
`sys.stderr.write(self._error_formatter.format(exc_type, exc_value, exc_stack) + os.linesep)` (line 405)
is already giving a good message.

Reply:
Alrighty! Is my commit okay?

Reply:
Not sure where these merge commits are coming from, are you doing it in gitbash / git gui or clicking a button in an ide?